cmd/cluster/cleanup_leaked_ec2.go

```diff
@@ -0,0 +1,181 @@
+package cluster
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/openshift/osdctl/pkg/k8s"
+	"github.com/openshift/osdctl/pkg/osdCloud"
+	"github.com/openshift/osdctl/pkg/utils"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/runtime"
+	capav1beta2 "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type cleanup struct {
+	awsClient   cleanupAWSClient
+	client      client.Client
+	cluster     *cmv1.Cluster
+	mgmtCluster *cmv1.Cluster
+
+	// ClusterId is the internal or external OCM cluster ID.
+	// This is optional, but typically is used to automatically detect the correct settings.
+	ClusterId string
+	// Yes default confirmation prompts to yes
+	Yes bool
+}
+
+type cleanupAWSClient interface {
+	DescribeInstances(ctx context.Context, params *ec2.DescribeInstancesInput, optFns ...func(options *ec2.Options)) (*ec2.DescribeInstancesOutput, error)
+	TerminateInstances(ctx context.Context, params *ec2.TerminateInstancesInput, optFns ...func(options *ec2.Options)) (*ec2.TerminateInstancesOutput, error)
+}
+
+func newCmdCleanupLeakedEC2() *cobra.Command {
+	c := &cleanup{}
+
+	cleanupCmd := &cobra.Command{
+		Use:   "cleanup-leaked-ec2",
+		Short: "Remediate impact of https://issues.redhat.com/browse/OCPBUGS-23174",
+		Example: `
+  # Run against a given ROSA HCP cluster
+  osdctl cluster cleanup-leaked-ec2 --cluster-id ${CLUSTER_ID}
+
+  # Assess all "error" state ROSA HCP clusters for impact
+  for cluster in $(ocm list cluster -p search="hypershift.enabled='true' and state='error'" --columns='id' --no-headers);
+  do
+    osdctl cluster cleanup-leaked-ec2 --cluster-id ${cluster}
+  done
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return c.Run(context.Background())
+		},
+	}
+
+	cleanupCmd.Flags().StringVarP(&c.ClusterId, "cluster-id", "C", "", "OCM internal/external cluster id to check for impact of OCPBUGS-23174.")
+	cleanupCmd.Flags().BoolVarP(&c.Yes, "yes", "y", false, "(optional) Skip confirmation prompt when terminating instances")
```

Comment:
Could we use "-S"/"--skip" or something similar for this flag? I think that's what we use elsewhere throughout osdctl and it'd be nice to have consistency.

Reply:
Oh `-S` is a global flag https://github.com/openshift/osdctl/blob/845f1e8d38660ace6cc2c45d48fe57d1307af36a/internal/utils/globalflags/globalflags.go#L20, so I didn't want to override it here (not sure if it's possible either, haven't tried)

but `-y` is used elsewhere in this repo, like with sending SL's https://github.com/openshift/osdctl/blob/845f1e8d38660ace6cc2c45d48fe57d1307af36a/cmd/servicelog/post.go#L84